### PR TITLE
feat: Implementation of ProjectRepo

### PR DESCRIPTION
**Pull request by iorisa**
**Features**
- Wrapper for GitRepository and FileRepository of project. Implementation of Chapter 4.6 of https://deepwisdom.feishu.cn/wiki/CUK4wImd7id9WlkQBNscIe9cnqh
- fixbug: all unit tests. excludes:
  - tests/metagpt/test_config.py
  - tests/metagpt/actions/test_action_node.py
  - tests/metagpt/actions/test_research.py
  - tests/metagpt/tools/test_search_engine.py
  - tests/metagpt/tools/test_web_browser_engine.py
  - tests/metagpt/utils/test_repair_llm_raw_output.py

### CODE REVIEW

File: metagpt/actions/write_teaching_plan.py [hunk 1]
```diff
@@ -24,7 +23,7 @@ async def run(self, with_message=None, **kwargs):
         statement_patterns = TeachingPlanBlock.TOPIC_STATEMENTS.get(self.topic, [])
         statements = []
         for p in statement_patterns:
-            s = self.format_value(p)
+            s = self.format_value(p, options=self.context.options)
```
Comment (geekan) [inline metagpt/actions/write_teaching_plan.py:26]: how about not using options?

File: metagpt/tools/openai_text_to_embedding.py [hunk 2]
```diff
@@ -43,12 +42,12 @@ class Config:
 
 
 class OpenAIText2Embedding:
```
Comment (geekan) [inline metagpt/tools/openai_text_to_embedding.py:44]: Is it possible to use OpenAI's aclient directly?

File: metagpt/actions/write_teaching_plan.py (hunk 1 above)
Comment (iorisa) [reply metagpt/actions/write_teaching_plan.py]: `Context.options` has been deleted

File: metagpt/tools/openai_text_to_embedding.py (hunk 2 above)
Comment (iorisa) [reply metagpt/tools/openai_text_to_embedding.py:44]: The official website did not provide information, so I didn't dare to use it.

**Review by geekan — APPROVED** [review]
LGTM

**Pull request merged**